jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/applications/ServerlessExplorerNodes.kt

```diff
@@ -47,5 +51,42 @@ class DeleteApplicationAction : DeleteResourceAction<ServerlessApplicationNode>(
     override fun performDelete(selected: ServerlessApplicationNode) {
         val client: CloudFormationClient = AwsClientManager.getInstance(selected.nodeProject).getClient()
         client.deleteStack { it.stackName(selected.stackName) }
+        waitForStackDeletionComplete(client, selected.stackName)
+    }
+
+    companion object {
+
+        // TODO Use Java SDK waiter once supported
+        fun waitForStackDeletionComplete(client: CloudFormationClient, stackName: String) {
+            waitForStackDeletionComplete(client, stackName, 120, 30)
+        }
+
+        @TestOnly
+        fun waitForStackDeletionComplete(client: CloudFormationClient, stackName: String, maxAttempts: Int, delay: Int) {
```

Comment:
This logic should probably be moved into a separate class so that it can be unit tested independently - then we wouldn't need this `@TestOnly` tag.

Reply:
I thought we are gonna remove this when Java SDK support waiter. But it's cleaner to move this away. Moved to the Cloudformation.kt file.